sale_order_type/models/account_move.py

```diff
@@ -12,26 +12,25 @@ class AccountMove(models.Model):
         string="Sale Type",
         compute="_compute_sale_type_id",
         store=True,
-        readonly=True,
-        states={"draft": [("readonly", False)]},
+        readonly=False,
+        states={"posted": [("readonly", True)], "cancel": [("readonly", True)]},
```

Comment:
Is there a benefict in doing it like this?
It's basically the same but reversed, no? :thinking:

Reply:
Odoo does not correctly manage read-only states in certain cases with computed writable fields. So if we set True to the read-only parameter, the value is not saved.
That is why the logic has been reversed, so that the value of the readonly attribute in the field is False and works correctly.
When Odoo fixes the problem the logic can be simplified again.

Reply:
Not sure if the problem is an Odoo bug or a mangling in inheritance of a mix of attrs and states, but anyway, please use `force_save="1"` in the view instead of changing this and everything solved.

Reply:
Please don't change this and use `force_save="1"` as stated

Reply:
`force_save="1"` works at JS view level and the issue is that the modifier is not keep in create method https://github.com/odoo/odoo/blob/27257d34c8b7374bfd8037de71f175f4ecb2913e/addons/web/static/src/js/views/basic/basic_model.js#L3690

`force_save="1"` don't work because `_fields` method:
https://github.com/odoo/odoo/blob/ed89c098b35405cf7c1da98f125761bb354e1297/odoo/models.py#L3701
gets field model definition and this evaluation:
https://github.com/odoo/odoo/blob/ed89c098b35405cf7c1da98f125761bb354e1297/odoo/models.py#L3722
avoid that the field value are protected.

Reply:
I finally got it: this is not a computed writable field for ORM if not:

- store=True
- readonly=False
- with compute method

so it's clear now and it doesn't matter that the condition is longer this way. It must be this.

Thanks for the patience and the explanations.